csrc/host_ir/jit.cpp

```diff
@@ -512,6 +532,185 @@ void inferTensorShapesAndStrides(
   NVF_ERROR_EQ(std::ssize(strides), logical_ndims);
 }
 
+// Helper function to infer allocation domain sizes and strides in LLVM IR
+// For now, implements Option A: assumes allocation domain is permutation of
+// logical (common case for views/permutes). Can be extended to handle
+// split/merge later.
+void inferAllocationSizesAndStridesLLVM(
```

Comment:
Can this somehow be merged with https://github.com/NVIDIA/Fuser/blob/023a8408bbad38afc22e4c17de39aebdf6c2a948/csrc/host_ir/jit.cpp#L378?